Snyk scan may fail because Nexus is not configured properly

**Describe the bug**
E.g. for Gradle-based projects, the Nexus host/credentials will be read from the environment. That is not set inside the Snyk stage, leading to authentication issues when dependencies need to be pulled.

**Affected version:**
 - OpenShift: all
 - OpenDevStack: 1.2

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```
+ snyk monitor --org=foo --file=build.gradle --project-name=bar

Monitoring /tmp/workspace/foo-cd/foo-cd-bar-develop...

Gradle Error (short):
> Could not resolve all dependencies for configuration ':classpath'.
   > Authentication scheme 'all'(Authentication) is not supported by protocol 'file'

===== DEBUG INFORMATION START =====
gradle command: '/tmp/workspace/foo-cd/foo-cd-bar-develop/gradlew' snykResolvedDepsJson -q --build-file build.gradle --no-daemon -Dorg.gradle.parallel= -PonlySubProject=. -I /tmp/tmp-xyz-init.gradle

------------------------------------------------------------
Gradle 5.6.2
------------------------------------------------------------

Build time:   2019-09-05 16:13:54 UTC
Revision:     55a5e53d855db8fc7b0e494412fc624051a8e781

Kotlin:       1.3.41
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          11.0.5 (Oracle Corporation 11.0.5+10-LTS)
OS:           Linux 3.10.0-957.21.3.el7.x86_64 amd64



>>> command: '/tmp/workspace/foo-cd/foo-cd-bar-develop/gradlew' snykResolvedDepsJson -q --build-file build.gradle --no-daemon -Dorg.gradle.parallel= -PonlySubProject=. -I /tmp/tmp-xyz-init.gradle
>>> exit code: 1
>>> stdout:

>>> stderr:
Picked up JAVA_TOOL_OPTIONS: -XX:+UnlockExperimentalVMOptions -Dsun.zip.disableMemoryMapping=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'bar'.
> Could not resolve all dependencies for configuration ':classpath'.
   > Authentication scheme 'all'(Authentication) is not supported by protocol 'file'
```

FYI @stitakis

--- a/vars/stageScanForSnyk.groovy
+++ b/vars/stageScanForSnyk.groovy
@@ -15,7 +15,8 @@ def call(def context, def snykAuthenticationCode, def buildFile, def organisatio
   } else {
     stage('Snyk Security Scan') {
       withEnv(["SNYK_AUTHENTICATION_CODE=${snykAuthenticationCode}", "PROJECT_NAME=${context.targetProject}",
-               "COMPONENT_NAME=${context.componentId}", "BUILD_FILE=${buildFile}", "ORGANISATION=${organisation}"]) {
+               "COMPONENT_NAME=${context.componentId}", "BUILD_FILE=${buildFile}", "ORGANISATION=${organisation}",
+               "NEXUS_HOST=${context.nexusHost}", "NEXUS_USERNAME=${context.nexusUsername}", "NEXUS_PASSWORD=${context.nexusPassword}"]) {
         // Verify that snyk is installed
         def status = sh(script: "snyk version", returnStatus: true, label : "getting snyk version")
         if (status != 0) {

